### PR TITLE
Add Textarea component

### PR DESCRIPTION
**Pull request by alexwllms**
## What does this do?

Add a `Textarea` component to the project.

## What does it affect?

- Add a `Textarea` component to the project
- Add stories for the default, error and disabled states
- Add tests for the component

## Any debts to pay?

- Need to bare in mind that when integrating into the Agent Portal we'll need to add in some missing positional styles

## Review Checklist

- [x] Added and/or modified tests for affected areas
- [x] Squashed any extraneous commit messages
- [x] Ready to review
- [ ] Ready to merge

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   ],
   "scripts": {
     "test": "jest --coverage",
-    "test:watch": "jest --watch",
+    "test:watch": "jest --watch --coverage",
```
Comment (txwkx) [inline package.json]: is this PR up to date with `master`?

File: src/Box/Box.tsx [hunk 2]
```diff
@@ -7,7 +6,7 @@ import {space, SpaceProps} from '../utils/space';
 
 interface IBox extends SpaceProps, FlexProps, MeasureProps {}
 
-export const Box: FC = styled.div<IBox>`
+export const Box = styled.div<IBox>`
```
Comment (txwkx) [inline src/Box/Box.tsx]: same comment as above

File: src/Textarea/Textarea.stories.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  .addWithJSX('default', () => (
+    <Textarea
+      id="textarea_id"
+      label={text('label', 'Textarea label')}
+      value={text('value', '')}
+      resize="both"
+      onChange={() => {}}
+      disabled={boolean('disabled', false)}
+      placeholder="Here is some placeholder text."
+      className={text('Class name', '')}
```
Comment (txwkx) [inline src/Textarea/Textarea.stories.js]: this **props** is required by `styled-components` only
I don't think we would have any use of explicitly setting it in the storybook

File: src/Textarea/Textarea.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  /** used for label - input connection */
+  name?: string;
+  /** input value */
+  value: string;
+  /** error flag */
+  error?: boolean;
+  /** error text message */
+  errorMsg?: string;
+  /** Allow user to resize the textarea vertically and horizontally or not */
+  resize: 'none' | 'both';
```
Comment (txwkx) [inline src/Textarea/Textarea.tsx]: shall we set `resize` as an optional prop that defaults to ***none***?

the number of use cases of users adjusting the size in practice is extremely low (personal assumption)

File: package.json (hunk 1 above)
Comment (alexwllms) [reply package.json]: Updated

**Pull request merged**